lib/PgSQLFFTO.cpp

```
@@ -0,0 +1,176 @@
+        if (is_select) report_query_stats(m_current_query, duration, 0, rows);
+        else report_query_stats(m_current_query, duration, rows, 0);
+        m_state = IDLE;
+    } else if (type == 'Z' || type == 'E') {
+        // ReadyForQuery or ErrorResponse. We don't always want to report stats here if 'C' already did it.
+        // But if we didn't get 'C', report what we have.
+        if (m_state == AWAITING_RESPONSE) {
+            unsigned long long duration = monotonic_time() - m_query_start_time;
+            report_query_stats(m_current_query, duration);
+        }
+        m_state = IDLE;
+    }
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)